src/mrinufft/trajectories/utils.py

```diff
@@ -173,69 +177,220 @@ class VDSpdf(StrEnum):
+
+    Parameters
+    ----------
+    gmax : float
+        Maximum gradient amplitude in T/m. Defaults to 40 mT/m.
+    smax : float
+        Maximum slew rate in T/m/s. Defaults to 200 T/m/s.
+    n_coils : int
+        Number of coils used in the MRI system. Defaults to 8.
+    dwell_time : float
+        Dwell time in seconds. Defaults to 1 ns.
+    grad_raster_time : float
+        Gradient raster time in seconds. Defaults to 5 us.
+    field_strength : float
+        Magnetic field strength in Tesla. Defaults to 3.0 T.
+
+    Attributes
+    ----------
+    raster_time : float
+        The gradient raster time in seconds, which is the same as `grad_raster_time`.
+
+    Notes
+    -----
+    The `Hardware` class encapsulates the hardware constraints for MRI sequences,
+    including the maximum gradient amplitude, slew rate, number of coils, dwell time,
+    gradient raster time, and magnetic field strength.
+    It is designed to be used in conjunction with the `Acquisition` class, which defines
+    acquisition parameters such as field of view, image size, and gyromagnetic ratio.
+
+    The default values for the parameters are set to typical values used in MRI systems.
+    These values can be adjusted based on the specific hardware being used for MRI
+    acquisition.
+
+    For convenience, we define several common hardware configurations,
+
+    """
+
+    gmax: float = 40 * SI.milli  # Maximum gradient amplitude in T/m
+    smax: float = 200  # T/m/s
+    n_coils: int = 32
+    dwell_time: float = 1 * SI.nano  # s
+    grad_raster_time: float = 5 * SI.micro  # s
+    field_strength: float = 3.0  # Tesla
+
+    @property
+    def raster_time(self) -> float:
+        """Alias of grad_raster_time."""
+        return self.grad_raster_time
+
+
+# fmt: off
+class SIEMENS:
+    """Common hardware configurations for Siemens MRI systems."""
+
+    TERRA          = Hardware(gmax=80*SI.milli, smax=200, field_strength=7)
+    TERRAX         = Hardware(gmax=200*SI.milli, smax=200, field_strength=7)
+    TERRAX_IMPULSE = Hardware(gmax=200*SI.milli, smax=900, field_strength=7)
+    PRISMA         = Hardware(gmax=40*SI.milli, smax=200, field_strength=3)
+    CIMA           = Hardware(gmax=200*SI.milli, smax=200, field_strength=3)
+    CIMAX          = Hardware(gmax=200*SI.milli, smax=200, field_strength=3)
+
+# fmt: on
+
+
+@dataclass(frozen=True)
+class Acquisition:
+    """
+    Acquisition configuration for MRI sequences.
+
+    Parameters
+    ----------
+    fov : tuple[float, float, float]
+        Field of View in meters (x, y, z).
+    img_size : tuple[int, int, int]
+        Image size in pixels (x, y, z).
+    hardware : Hardware
+        Hardware configuration for the acquisition.
+    gamma : Gammas, optional
+        Gyromagnetic ratio in Hz/T for the nucleus being imaged.
+        Defaults to `Gammas.HYDROGEN`.
+    oversampling : int, optional
+        Oversampling factor for the ADC. Defaults to 1.
```

Comment:
Not really needed if you have dwell time in hardware config. Also, I feel dwell time should be here rather than in "hardware config", as it is not specific to a scanner type but rather a scan itself.

Reply:
Well the constructor could either take oversampling or dwell_time at init, we have a similar issue with resolution as well. I think some getter/setter wizardry could do the trick.

Reply:
The thing is oversampling was originally defined from sparkling, but i want to get away from its use, we just sample at higher rate, it's not "over" sampled , which generally leads people to believe that you are above nyquist rate by that factor